db/database.go

```diff
@@ -377,6 +395,16 @@ func NewDatabaseContext(ctx context.Context, dbName string, bucket base.Bucket,
 		CollectionByID: make(map[uint32]*DatabaseCollection),
 	}
 
+	// Initialize metadata ID and keys
+	if options.Scopes.onlyDefaultCollection() {
+		dbContext.MetadataID = ""
+	} else {
+		// TODO: apply length limit to MetadataPrefix
+		dbContext.MetadataID = dbName
+	}
+	metaKeys := base.NewMetadataKeys(dbContext.MetadataID)
+	dbContext.MetadataKeys = metaKeys
```

Comment:
I don't think `dbContext.MetadataID` needs to exist given it's being stored in `MetadataKeys` - unless I'm overlooking something not yet implemented that requires it be separate.